src/app/util/ember-compatibility-functions.cpp

```diff
@@ -37,6 +47,72 @@ EmberAfClusterCommand imCompatibilityEmberAfCluster;
+    case ZCL_COMMAND_ID_ATTRIBUTE_TYPE:  // Command Id
+    case ZCL_BITMAP8_ATTRIBUTE_TYPE:     // 8-bit bitmap
+    case ZCL_ENUM8_ATTRIBUTE_TYPE:       // 8-bit enumeration
+    case ZCL_INT8U_ATTRIBUTE_TYPE:       // Unsigned 8-bit integer
+        static_assert(std::is_same<chip::EndpointId, uint8_t>::value,
+                      "chip::EndpointId is expected to be uint8_t, change this when necessary");
+        static_assert(std::is_same<chip::CommandId, uint8_t>::value,
+                      "chip::CommandId is expected to be uint8_t, change this when necessary");
+        return ZCL_INT8U_ATTRIBUTE_TYPE;
+
+    case ZCL_GROUP_ID_ATTRIBUTE_TYPE: // Group Id
+    case ZCL_ENUM16_ATTRIBUTE_TYPE:   // 16-bit enumeration
+    case ZCL_BITMAP16_ATTRIBUTE_TYPE: // 16-bit bitmap
+    case ZCL_INT16U_ATTRIBUTE_TYPE:   // Unsigned 16-bit integer
+        static_assert(std::is_same<chip::GroupId, uint16_t>::value,
+                      "chip::GroupId is expected to be uint8_t, change this when necessary");
+        return ZCL_INT16U_ATTRIBUTE_TYPE;
+
+    case ZCL_BITMAP32_ATTRIBUTE_TYPE:       // 32-bit bitmap
+    case ZCL_DEVICE_TYPE_ID_ATTRIBUTE_TYPE: // Device Type Id
+    case ZCL_INT32U_ATTRIBUTE_TYPE:         // Unsigned 32-bit integer
+        static_assert(std::is_same<chip::DeviceTypeId, uint32_t>::value,
+                      "chip::DeviceTypeId is expected to be uint32_t, change this when necessary");
+        return ZCL_INT32U_ATTRIBUTE_TYPE;
+
+    case ZCL_BITMAP64_ATTRIBUTE_TYPE:  // 64-bit bitmap
+    case ZCL_FABRIC_ID_ATTRIBUTE_TYPE: // Fabric Id
+    case ZCL_NODE_ID_ATTRIBUTE_TYPE:   // Node Id
+    case ZCL_INT64U_ATTRIBUTE_TYPE:    // Unsigned 64-bit integer
+        static_assert(std::is_same<chip::FabricId, uint64_t>::value,
+                      "chip::FabricId is expected to be uint64_t, change this when necessary");
+        static_assert(std::is_same<chip::NodeId, uint64_t>::value,
+                      "chip::NodeId is expected to be uint64_t, change this when necessary");
+        return ZCL_INT64U_ATTRIBUTE_TYPE;
+
+    case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
+        return ZCL_INT8S_ATTRIBUTE_TYPE;
+    case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
+        return ZCL_INT16S_ATTRIBUTE_TYPE;
+    case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
+        return ZCL_INT32S_ATTRIBUTE_TYPE;
+    case ZCL_INT64S_ATTRIBUTE_TYPE: // Signed 64-bit integer
+        return ZCL_INT64S_ATTRIBUTE_TYPE;
+
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE: // Octet string
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE:  // Character string
+        return ZCL_OCTET_STRING_ATTRIBUTE_TYPE;
+
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE:
+        return ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE;
+
+    case ZCL_ARRAY_ATTRIBUTE_TYPE:
+        return ZCL_ARRAY_ATTRIBUTE_TYPE;
+    default:
+        return type;
```

Comment:
Since there is a `default` case, is there any reason to not let the `case` that return the same type falls into `default` ? That would make `BaseType` a list of special cases with a default handler.

Reply:
Fixed